exec/java-exec/src/main/java/org/apache/drill/exec/store/schedule/BlockMapBuilder.java

```diff
@@ -104,12 +104,16 @@ public BlockMapReader(FileStatus status, boolean blockify) {
     @Override
     protected List<CompleteFileWork> runInner() throws Exception {
       final List<CompleteFileWork> work = Lists.newArrayList();
+
+      final Set<String> noDrillbitHosts = logger.isDebugEnabled() ? Sets.<String>newHashSet() : null;
```

Comment:
Consider moving `noDillbitHosts` to `BlockMapBuilder` class (use `Sets.newConcurrentHashSet()` in this case) as it does not seem to belong to `BlockMapReader`. With such change, other changes are not necessary and likely this will allow reducing repeated log messages even further. Drop `<Sring>` from `Sets.<String>newHashSet()`.